### PR TITLE
WIP: changed verifyGroupDataMatchesWithFs to be non-blocking.

### PR DESCRIPTION
**Pull request by PhotoNomad0**
#### Describe what your pull request addresses (Please include relevant issue numbers):
- changed verifyGroupDataMatchesWithFs to be non-blocking.

#### Please include detailed Test instructions for your pull request:
-

#### Standard Test Instructions for PR Review Process:

- [ ] Double check unit tests that have been written
- [ ] Check for documentation for code changes
- [ ] Check that there are not inadvertent commits to tC Apps when reviewing a tC Core PR
- [ ] Checkout the branch locally and ensure that app runs as expected
  - [ ] Ensure tests pass
  - [ ] Open and watch the console for errors
  - [ ] Make sure all actions perform as expected
  - [ ] Import and Load a new Project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Switch project to an existing project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Next time reverse the order of importing after loading an existing project
- [ ] Reviewer should double check the DoD in the ISSUE, including the “spirit” of the story
- [ ] Ask Ben or Koz if you have any concerns about implementation (especially UI related)

---
This change is [Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6442)

### CODE REVIEW

**Comment by PhotoNomad0** [comment]
non-blocking verifyGroupDataMatchesWithFsNonBlocking actually runs slower:

**blocking:**

- GroupsDataActions.verifyGroupDataMatchesWithFs(wordAlignment, /Users/blm/translationCore/projects/en_ultx_mat_book) took 1440.4999999969732ms 
-openTool(wordAlignment) took 4596.300000004703ms 


- GroupsDataActions.verifyGroupDataMatchesWithFs(translationWords, /Users/blm/translationCore/projects/en_ultx_mat_book) took 3049.9000000054366ms 
- openTool(translationWords) took 6678.599999999278ms 



**non-blocking**

- GroupsDataActions.verifyGroupDataMatchesWithFsNonBlocking(wordAlignment, /Users/blm/translationCore/projects/en_ultx_mat_book) took 1747.3000000027241ms 
- openTool(wordAlignment) took 4937.900000004447ms 

- GroupsDataActions.verifyGroupDataMatchesWithFsNonBlocking(translationWords, /Users/blm/translationCore/projects/en_ultx_mat_book) took 3628.1999999991967ms 
- openTool(translationWords) took 7151.700000002165ms

File: __mocks__/fs-extra.js [hunk 1]
```diff
@@ -44,6 +44,10 @@ function readdirSync(directoryPath) {
   return [];
 }
 
+function readdir(directoryPath) {
```
Comment (mannycolon) [inline __mocks__/fs-extra.js:47]: Nice Job adding these two functions

File: src/js/actions/GroupsDataActions.js [hunk 2]
```diff
@@ -175,6 +175,133 @@ export function verifyGroupDataMatchesWithFs() {
...
+
+        for ( let j = 0, lenC = chapters.length; j < lenC; j++) {
+          const chapterFolder = chapters[j];
+          const chapterDir = path.join(dataPath, chapterFolder);
+
+          if (!fs.existsSync(chapterDir)) {
+            continue;
+          }
+
+          let verses = fs.readdirSync(chapterDir);
```
Comment (mannycolon) [inline src/js/actions/GroupsDataActions.js]: This function is synchronous (UI blocking)

**Review by PhotoNomad0 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6442)* status: 0 of 6 files reviewed, 2 unresolved discussions (waiting on @mannycolon and @PhotoNomad0)

___
*[src/js/actions/GroupsDataActions.js, line 227 at r2](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6442#-Lp3qp-XbQsZmIJrWdIb-r2-227:-Lp4tmt14l2sBBXxkcm6:b-bb1jjp) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/37ff36eede8fe2124c9c18cc7bd1973d9321dcdd/src/js/actions/GroupsDataActions.js#L227)):*
<details><summary><i>Previously, mannycolon (Manny Colon) wrote…</i></summary><blockquote>

This function is synchronous (UI blocking)
</blockquote></details>

I had accidently pushed up changes to do minimal test.

**Review by PhotoNomad0 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6442)* status: 0 of 6 files reviewed, 2 unresolved discussions (waiting on @mannycolon and @PhotoNomad0)

___
*[src/js/actions/GroupsDataActions.js, line 227 at r2](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6442#-Lp3qp-XbQsZmIJrWdIb-r2-227:-Lp4u35D0nNavNZmbdeC:bbgd3n6) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/37ff36eede8fe2124c9c18cc7bd1973d9321dcdd/src/js/actions/GroupsDataActions.js#L227)):*
<details><summary><i>Previously, PhotoNomad0 (Bruce McLean) wrote…</i></summary><blockquote>

I had accidently pushed up changes to do minimal test.
</blockquote></details>

Moved to different branch.

**Comment by PhotoNomad0** [comment]
SLower

**Pull request closed without merging**